integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkRestIcebergIntegrationTest.java

```diff
@@ -0,0 +1,206 @@
+          .withExposedPorts(8181)
+          .withNetwork(network);
+
+  private static final String JAVA_VERSION = System.getProperty("java.version");
+
+  @BeforeAll
+  @SneakyThrows
+  public static void beforeAll() {
+    SparkSession$.MODULE$.cleanupAnyExistingSession();
+    FileUtils.deleteDirectory(new File("/tmp/iceberg/"));
+    mockServer = MockServerUtils.createAndConfigureMockServer(MOCK_SERVER_PORT);
+    icebergRestContainer.start();
+  }
+
+  @AfterAll
+  @SneakyThrows
+  public static void afterAll() {
+    SparkSession$.MODULE$.cleanupAnyExistingSession();
+    MockServerUtils.stopMockServer(mockServer);
+    icebergRestContainer.stop();
+  }
+
+  @BeforeEach
+  @SneakyThrows
+  public void beforeEach() {
+    MockServerUtils.clearRequests(mockServer);
+    spark =
+        SparkSession.builder()
+            .master("local[*]")
+            .appName("IcebergIntegrationTest")
+            .config("spark.driver.host", LOCAL_IP)
+            .config("spark.driver.bindAddress", LOCAL_IP)
+            .config("spark.ui.enabled", false)
+            .config("spark.sql.shuffle.partitions", 1)
+            .config("spark.driver.extraJavaOptions", "-Dderby.system.home=/tmp/iceberg")
+            .config("spark.openlineage.transport.type", "http")
+            .config(
+                "spark.openlineage.transport.url",
+                "http://localhost:" + mockServer.getPort() + "/api/v1/namespaces/iceberg-namespace")
+            .config("spark.extraListeners", OpenLineageSparkListener.class.getName())
+            .config("spark.sql.catalog.rest", "org.apache.iceberg.spark.SparkCatalog")
+            .config("spark.sql.catalog.rest.catalog-impl", "org.apache.iceberg.rest.RESTCatalog")
+            .config(
+                "spark.sql.catalog.rest.uri",
+                "http://"
+                    + icebergRestContainer.getHost()
+                    + ":"
+                    + icebergRestContainer.getMappedPort(8181))
+            .config(
+                "spark.sql.extensions",
+                "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions")
+            .getOrCreate();
+  }
+
+  private Dataset<Row> createTempDataset(int rows) {
+    List<Row> rowList =
+        Arrays.stream(IntStream.rangeClosed(1, rows).toArray())
+            .mapToObj(i -> RowFactory.create((long) i, (long) i + 1))
+            .collect(Collectors.toList());
+
+    return spark
+        .createDataFrame(
+            rowList,
+            new StructType(
+                new StructField[] {
+                  new StructField("a", LongType$.MODULE$, false, Metadata.empty()),
+                  new StructField("b", LongType$.MODULE$, false, Metadata.empty())
+                }))
+        .repartition(1);
+  }
+
+  @Test
+  @SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert")
+  void testMetricsFacets() {
+    if (JAVA_VERSION.startsWith("1.8") && System.getProperty(SPARK_VERSION).startsWith("3.5")) {
```

Comment:
nit: use EnabledIfSystemProperty or EnabledIfEnvironmentVariable for those checks instead of doing it inside test